addons/rmsmartshape/shapes/shape.gd

```diff
@@ -1118,42 +1118,39 @@ static func build_quad_corner(
 ) -> SS2D_Quad:
 	var new_quad := SS2D_Quad.new()
 
-	var quad_size: Vector2 = size / 2.0
+	#             :BUILD PLAN:
+	#   OUTER CORNER        INNER CORNER
+	#
+	#   0------A-----D             0-----0
+	#   |  1   :  2  |             |  3  :
+	#   0......B.....C             |     :
+	#          :     |     0-------D-----A
+	#          :  3  |     |  1    |  2  :
+	#          0-----0     0.......C.....B
+	#
+	#  1-previous, 2-current, 3-next (points)
+
+	var quad_size: Vector2 = size * 0.5
 	var delta_12: Vector2 = pt - pt_prev
 	var delta_23: Vector2 = pt_next - pt
-	var normal_23 := Vector2(delta_23.y, -delta_23.x).normalized()
-	var normal_12 := Vector2(delta_12.y, -delta_12.x).normalized()
+	var offset_12: Vector2 = delta_12.normalized() * custom_scale * pt_width * quad_size
+	var offset_23: Vector2 = delta_23.normalized() * custom_scale * pt_prev_width * quad_size
+	var custom_offset_13: Vector2 = (offset_12.normalized() - offset_23.normalized()) * custom_offset * quad_size
```

Comment:
`offset_12` and `offset_23` should have the same normalized vector as `delta_12` and `delta_23`, as they are just scaled versions of them.
Please cache them, easy micro-optimization.